cpp/src/io/utilities/datasource.cpp

```diff
@@ -470,7 +471,16 @@ class remote_file_source : public datasource {
  private:
   kvikio::RemoteHandle _kvikio_file;
 };
-
+#else
+/**
+ * @brief When KvikIO remote IO is disabled, `is_supported_remote_url()` return false always.
+ */
+class remote_file_source : public file_source {
+ public:
+  explicit remote_file_source(char const* filepath) : file_source(filepath) {}
+  static constexpr bool is_supported_remote_url(std::string const& url) { return false; }
```

Comment:
Remove parameter name to avoid unused warning.
```suggestion
  static constexpr bool is_supported_remote_url(std::string const&) { return false; }
```